backend/core/socketio_config.py

```diff
@@ -202,7 +204,84 @@ def handle_incoming_message(session: Session, message: dict):
     session.flush()
 
     user = user.serialize()
-    return user["id"], user["phone_number"]
+    return user["id"], user["phone_number"], chat_session_id, new_chat.id
+
+
+def update_chat_status(session: Session, message: dict):
+    conversation_envelope = get_value_or_raise_error(
+        message, "conversation_envelope"
+    )
+    chat_id = conversation_envelope.get("message_id", None)
+    logger.info(f"Updating chat {chat_id} status")
+    chat = session.exec(select(Chat).where(Chat.id == chat_id)).first()
+    if not chat:
+        return None
+    chat.status = 1
+    session.commit()
+    session.flush()
+    logger.info(f"Chat {chat_id} status updated")
+
+
+async def resend_messages(session: Session, user_id=int, user_sid=str):
+    chat_session = session.exec(
+        select(Chat_Session).where(Chat_Session.user_id == user_id)
+    ).all()
+    if not chat_session:
+        return None
+    last_ten_chats = session.exec(
+        select(Chat)
+        .where(
+            and_(
+                Chat.chat_session_id.in_([cs.id for cs in chat_session]),
+                Chat.sender_role.in_(
+                    [Sender_Role_Enum.CLIENT, Sender_Role_Enum.ASSISTANT]
+                ),
+            )
+        )
+        .order_by(Chat.created_at.desc())
+        .limit(10)
+    ).all()
+    # Reorder the results by created_at in ascending order
+    last_ten_chats = sorted(last_ten_chats, key=lambda x: x.created_at)
```

Comment:
Sorting should be a client concern, not a server concern. The client still has to deduplicate with the messages it already has. Please move sorting to the client.